### ISSUE TITLE
sort-comp does not detect component declaration starting with `return`

### ISSUE THREAD

**Issue by webOS101** [yannickcr/eslint-plugin-react#1076]
Similar to #174 a component is not detected if the declaration begins with `return` (Such as, in our case, where we're wrapping the component inside a HOC).  Other react rules appear to be applied correctly to these cases, just not `sort-comp`.

Example code (does not show `sort-comp` warning):
```
let makeComp = () => {
	return class MyComponent extends React.Component {
		componentDidUpdate () { }
		componentDidMount () {}
		constructor () { super(); }
	};
};
```

Example code (correctly shows `sort-comp` warning):
```
let makeComp = () => {
	class MyComponent extends React.Component {
		componentDidUpdate () { }
		componentDidMount () {}
		constructor () { super(); }
	};
	return MyComponent;
};
```

As another note:  The react lifecycle methods are out-of-order and yet generate no warning.  Not sure why they are not, maybe that's another ticket.

**Comment by ljharb** [comment]
Since `makeComp` is not a component, it's an HOC, are you saying that `MyComponent` isn't being treated as a React component in the second example?

**Comment by webOS101** [comment]
In the second example, it is correctly being interpreted as a React component.  In the first example, it is not detected as a React component.

**Comment by ljharb** [comment]
aha, thank you.
